conans/model/ref.py

```diff
@@ -58,7 +59,21 @@ def __new__(cls, name, version, user, channel):
         ConanName.validate_name(user)
         ConanName.validate_name(channel)
         version = Version(version)
-        return super(cls, ConanFileReference).__new__(cls, name, version, user, channel)
+        obj = super(cls, ConanFileReference).__new__(cls, name, version, user, channel)
+        obj.revision = revision
+        return obj
+
+    def __eq__(self, value):
+        if not value:
+            return False
+        return super(ConanFileReference, self).__eq__(value) and self.revision == value.revision
+
+    def __ne__(self, other):
+        """Overrides the default implementation (unnecessary in Python 3)"""
+        return not self.__eq__(other)
+
+    def __hash__(self):
```

Comment:
Because (If I remember correctly) it is used in the comparison, and by default, I don't want to compare the revision unless I told it explicitly.

Reply:
It is not used in comparisons, it is used in sets and dicts. If expected many collisions of different revisions of same ConanFileReference in some dict, then, it would make sense to include the revision in the hash. I don't think this is a case in the codebase

Reply:
Now I'm confused. `return hash((self.name, self.version, self.user, self.channel, self.revision))` is not hashing the revision?

Reply:
Sorry, my bad. I misread the code.